common/addons/library-image/src/main/java/com/dfsek/terra/addons/image/colorsampler/ColorSampler.java

```diff
@@ -0,0 +1,12 @@
+package com.dfsek.terra.addons.image.colorsampler;
+
+@FunctionalInterface
+public interface ColorSampler {
+    
+    /**
+     * @param x World x coordinate
+     * @param z World z coordinate
+     * @return Integer representing a web color
+     */
+    Integer apply(int x, int z);
```

Comment:
is there a reason this is a boxed type?

Reply:
ColorConverters were extending `Function<Integer, T>` so I kept boxed types here rather than autoboxing/unboxing since the color typically gets passed straight to the converter - changed that to just a regular interface so it all uses primitive `int` instead since `ColorConverter`  doesn't strictly need to be a functional interface